controllers/usersignupcleanup/usersignup_cleanup_controller.go

```diff
@@ -136,7 +139,27 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		// then delete the UserSignup
 		deactivatedThreshold := time.Now().Add(-time.Duration(config.Deactivation().UserSignupDeactivatedRetentionDays()*24) * time.Hour)
 
-		if cond.LastTransitionTime.Time.Before(deactivatedThreshold) {
+		meetsActivationCriteria := false
+		if !activationsAnnotationPresent {
+			// Meets the criteria if there is no activations annotation
+			meetsActivationCriteria = true
+		} else {
+			activationCount, err := strconv.Atoi(activations)
+			if err != nil {
+				// If there was an error converting the activation count to an integer, then assume it meets the criteria for deletion
+				meetsActivationCriteria = true
+			} else {
+				// Otherwise it meets the criteria if we can determine there has only ever been one activation in total
+				meetsActivationCriteria = activationCount <= 1
```

Comment:
Let's maybe move `1` to a const so it's easier to update in the feature. And down the road we might want to move it configuration. But for now it's fine to keep it as a const.

Reply:
Fixed